libraries/Microsoft.Bot.Builder.Ai/Translator.cs

```diff
@@ -82,15 +82,19 @@ public void AddNoTranslatePhrase(string noTranslatePhrase)
                     string[] wordIndexes = alignData.Split('-');
                     int srcStartIndex = Int32.Parse(wordIndexes[0].Split(':')[0]);
                     int srcLength = Int32.Parse(wordIndexes[0].Split(':')[1]) - srcStartIndex + 1;
+                    if ((srcLength + srcStartIndex) > sourceMessage.Length)
+                        continue;
                     string srcWrd = sourceMessage.Substring(srcStartIndex, srcLength);
-                    int sourceWordIndex = Array.FindIndex(srcWrds, row => row == srcWrd);
+                    int sourceWordIndex = Array.FindIndex(srcWrds, row => row.Contains(srcWrd));
```

Comment:
Lots of code changes in this area of code. 

However, there are NO relevant test changes. 

Please add test coverage to these changes.

Reply:
I have added this test Translator_PatternsTest in Translatortests Code